src/PrinterCartridgeLevelAlert.php

```diff
@@ -0,0 +1,383 @@
+                    echo "</ul>";
+                    echo "<div class='tab-content p-2 flex-grow-1 card border-start-0' style='min-height: 150px'>";
+                    echo "<div class='alltab'>";
+                    echo __('Cartridges whose level is low');
+                    echo "</div>";
+
+                    if (count($result) > 0) {
+                        if (Session::isMultiEntitiesMode()) {
+                            $nbcol = 4;
+                        } else {
+                            $nbcol = 3;
+                        };
+
+                        echo "<div class='flex-grow-1 d-flex flex-wrap flex-md-nowrap  align-items-center justify-content-between mb-2 search-pager'>";
+                        echo "<div class='table-responsive'>";
+                        echo "<table class='table table-hover' cellspacing='2' cellpadding='3'>";
+                        echo "<thead>";
+                        echo "<tr>";
+                        echo "<th>" . Printer::getTypeName(Session::getPluralNumber()) . "</th>";
+
+                        if (Session::isMultiEntitiesMode()) {
+                            echo "<th>" . Entity::getTypeName(Session::getPluralNumber()) . "</th>";
+                        }
+
+                        echo "<th>" . Cartridge::getTypeName(Session::getPluralNumber()) . "</th>";
+                        echo "<th>" . __('Level') . "</th>";
+                        echo "<th>" . _n('Email notification', 'Email notifications', false) . "</th>";
+                        echo "</tr>";
+                        echo "</thead>";
+
+                        foreach ($result as $data) {
+                            echo self::displayBody($data);
+                        }
+                        echo "</table>";
+                        echo "</div>";
+                        echo "</div>";
+                    } else {
+                        echo "<br><div align='center'><b>" . __('No cartridge is below the threshold') . "</b></div>";
+                    }
+                    echo "</div>";
+                    echo "</div>";
+                    echo "</div>";
+                    echo "</div>";
+                    echo "</div>";
+                }
+            }
+        }
+    }
+
+    /**
+     * Get task description
+     *
+     * @return string
+     */
+    public static function getTaskDescription(): string
+    {
+        return __("Notification sending for low level printer cartridges");
+    }
+
+    /**
+     *
+     * @return string
+     */
+    public static function cronInfo($name)
+    {
+        return [
+            'description' => self::getTaskDescription()
+        ];
+    }
+
+    /**
+     * Cron action on cartridges : alert if a stock is behind the threshold
+     *
+     * @param CronTask $task CronTask for log, display information if NULL? (default NULL)
+     *
+     * @return void
+     **/
```

Comment:
PHPDoc description seems wrong for this method.
Param type should be CronTask|null since it can accept null.
Return type is declared to be void but the method always returns 0.

Reply:
I simply copied/pasted some code from cronCartridge and it seems to have the same errors :) . Some mess in cronConsumable also... Will change mine :)